player/external_files.c

```diff
@@ -110,36 +110,52 @@ static int compare_sub_priority(const void *a, const void *b)
     return strcoll(s1->fname, s2->fname);
 }
 
-static struct bstr guess_lang_from_filename(struct bstr name, int *fn_start)
+static struct bstr guess_lang_from_filename(struct bstr name, int *lang_start)
 {
     if (name.len < 2)
         return (struct bstr){NULL, 0};
 
-    int n = 0;
+    int lang_length = 0;
     int i = name.len - 1;
 
-    char thing = '.';
+    char delimiter = '.';
     if (name.start[i] == ')') {
-        thing = '(';
+        delimiter = '(';
         i--;
     }
     if (name.start[i] == ']') {
-        thing = '[';
+        delimiter = '[';
         i--;
     }
 
     while (i >= 0 && mp_isalpha(name.start[i])) {
-        n++;
-        if (n > 3)
-            return (struct bstr){NULL, 0};
+        lang_length++;
         i--;
     }
 
-    if (n < 2 || i == 0 || name.start[i] != thing)
+    if (lang_length < 2 || lang_length > 6 || (name.start[i] != '-' && lang_length > 3))
```

Comment:
`gsw-u-sd-chzh` is longer than 6 chars

Reply:
Here's a more extreme example: `he-IL-u-ca-hebrew-tz-jeruslm`, The only way to add full support for the BCP 47 language tags standard to external files seems to be to remove the limit on the detection length.

Reply:
One possible detection method: invalid when there are other symbols other than `-`. According to the BCP 47 language tags standard, there will be a maximum of eight characters after each `-` symbol.

Reply:
I just removed the limit.